packages/core/src/data-editor/data-editor.tsx

```diff
@@ -2655,7 +2761,12 @@ const DataEditorImpl: React.ForwardRefRenderFunction<DataEditorRef, DataEditorPr
     const scrollToRef = React.useRef(scrollTo);
     scrollToRef.current = scrollTo;
     React.useEffect(() => {
-        if (outCol !== undefined && outRow !== undefined) {
+        if (
+            outCol !== undefined &&
+            outRow !== undefined &&
+            (outCol !== expectedExternalGridSelection.current?.current?.cell[0] ||
```

Comment:
This is probably too naughty of a hack. The better solution seemingly is to not pre-scroll when the user is controlling the gridSelection and always use this effect to perform any selection maintenance scrolling.

Reply:
I've opted to keep this for now but have added some extra testing.